hack/update-bazel.sh

```diff
@@ -38,10 +38,9 @@ rm -f ${TESTINFRA_ROOT}/vendor/k8s.io/apimachinery/pkg/util/sets/BUILD
   e26fc85d14a1d3dc25569831acc06919673c545a \
   "${TMP_GOPATH}"
 
-# The gazelle commit should match the rules_go commit in the WORKSPACE file.
 "${TESTINFRA_ROOT}/hack/go_install_from_commit.sh" \
-  github.com/bazelbuild/rules_go/go/tools/gazelle/gazelle \
-  0.8.1 \
+  github.com/bazelbuild/bazel-gazelle/cmd/gazelle \
+  0.8 \
```

Comment:
for anyone following along: https://github.com/bazelbuild/bazel-gazelle/releases/tag/0.8
> This is the first release of Gazelle in its new repository. This matches the Gazelle from rules_go at version 0.8.1, hence the similar version number.